src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,52 @@
+.calendar {
+  position: relative;
+  padding: 10px;
+  display: flex;
+  flex-wrap: wrap;
+  width: $widthCall;
+
+  @each $dayName, $mgLeft in $days {
+    &--start-day--#{$dayName} {
+      & > :first-child {
+        margin-left: $mgLeft;
+      }
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &--month-length--#{$index} {
+      $dayIndex: $index + 1;
+      & > :nth-child(n + #{$dayIndex}) {
+        opacity: 0;
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    background-color: $colorCell;
+    box-sizing: border-box;
+    border: 1px solid black;
+    margin: 0.5px;
```

Comment:
We can use 'gap: 1px' in this task because of no test check here.
If you want to use margins, 'margin-right: 1px' and 'margin-bottom: 1px' would be better.